### PR TITLE
TDR-3234 Refactor API ValidateUserOwnsFiles authorisation tag 

### PR DESCRIPTION
**Pull request by tna-aford**
[TDR-3234](https://national-archives.atlassian.net/browse/TDR-3234?atlOrigin=eyJpIjoiMzMwNjFlY2FhZDVmNDg3OWIxZmU2OGRkYTFlZjM4ZDQiLCJwIjoiaiJ9)

[TDR-3234]: https://national-archives.atlassian.net/browse/TDR-3234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- [x] Altered methods to remove consignment id checks, instead checking for user ID
- [x] updated tests accordingly & test cases

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/auth/AuthorisationTags.scala [hunk 1]
```diff
@@ -182,20 +181,17 @@ case class ValidateUserOwnsFiles[T](argument: Argument[T]) extends Authorisation
...
-      filesThatDoNotBelongToAConsignment: Seq[UUID] = fileIds.filterNot(fileId => fileIdsBelongingToAConsignment.contains(fileId))
-      fileIdsThatDoNotBelongToTheUser: Seq[UUID] = fileIdsAndOwner.collect {
-        case fileIdAndOwner if fileIdAndOwner.userId != userId => fileIdAndOwner.fileId
+      fileOwner: Seq[(UUID, UUID)] <- ctx.ctx.fileService.getOwnersOfFiles(fileIds)
+      fileIdsThatDoNotBelongToTheUser: Seq[UUID] = fileOwner.collect {
+        case (fileId, ownerId) if ownerId != userId => fileId
       }
-      allFilesBelongToAConsignment = filesThatDoNotBelongToAConsignment.isEmpty
-      allFilesBelongToTheUser = fileIdsThatDoNotBelongToTheUser.isEmpty
+      fileIdsWithNoOwner: Seq[UUID] = fileIds.filterNot(fileId => fileOwner.exists(_._1 == fileId))
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/auth/AuthorisationTags.scala:188]: Looking the logic here again, it should be possible to simplify it a bit if you combine the `fileIdsThatDoNotBelongToTheUser` and `fileIdsWithNoOwner` into a single list.

This single list can then be used in the `if` statement, ie `if (list.isEmpty || exportAccess)`.

That single list can then be used in the error message, which will then mean the json responses will be correct, without the need to change them.

File: src/test/resources/json/addFileStatuses_data_invalid_fileid.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "data": null,
   "errors": [
     {
-      "message": "User 'dfee3d4f-3bb1-492e-9c85-7db1685ab12f' does not own the files they are trying to access:\na8dc972d-58f9-4733-8bb2-4254b89a35f2 or does not have export access",
+      "message": "User 'dfee3d4f-3bb1-492e-9c85-7db1685ab12f' does not own the files they are trying to access:\n or does not have export access",
```
Comment (TomJKing) [inline src/test/resources/json/addFileStatuses_data_invalid_fileid.json:5]: If the logic in the tag is changed to use a single combined list, this response does not need changing. See comments above.

File: src/test/resources/json/updatebulkfilemetadata_data_fileid_not_exists.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "data": null,
   "errors": [
     {
-      "message": "User '4ab14990-ed63-4615-8336-56fbb9960300' does not own the files they are trying to access:\n373ce1c5-6e06-423d-8b86-ca5eaebef457\n5302acac-1396-44fe-9094-dc262414a03a or does not have export access"
+      "message": "User '4ab14990-ed63-4615-8336-56fbb9960300' does not own the files they are trying to access:\n or does not have export access"
```
Comment (TomJKing) [inline src/test/resources/json/updatebulkfilemetadata_data_fileid_not_exists.json:5]: If the logic in the tag is changed to use a single combined list, this response does not need changing. See comments above.

**Review by TomJKing — COMMENTED** [review]
Looks good. Added some comments. Thanks

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala [hunk 4]
```diff
@@ -254,6 +254,65 @@ class FileRepositorySpec extends TestContainerUtils with ScalaFutures with Match
...
+
+    val files = fileRepository.getFileFields(fileIds).futureValue
+
+    files.size shouldBe 2
+    files.head._1 shouldBe fileOneId
+    files(1)._1 shouldBe fileTwoId
+    files.head._2 shouldBe userId
+    files(1)._2 shouldBe userId
+  }
+  "getFileFields" should "return one file if given duplicate fileIds" in withContainers { case container: PostgreSQLContainer =>
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala:278]: Missing line break between tests

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala [hunk 5]
```diff
@@ -254,6 +254,65 @@ class FileRepositorySpec extends TestContainerUtils with ScalaFutures with Match
...
+    val fileIds = Seq(fileOneId, fileTwoId)
+
+    val files = fileRepository.getFileFields(fileIds).futureValue
+
+    files.size shouldBe 1
+    files.head._1 shouldBe fileOneId
+    println(s" files : ${files.size}, ${files}")
+    files.head._2 shouldBe userId
+  }
+  "getFileFields" should "return an empty Seq if given no fileID" in withContainers { case container: PostgreSQLContainer =>
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala:297]: Missing line break between tests

File: src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala [hunk 6]
```diff
@@ -254,6 +254,65 @@ class FileRepositorySpec extends TestContainerUtils with ScalaFutures with Match
...
+
+    utils.createConsignment(consignmentId, userId)
+    utils.createFile(fileOneId, consignmentId)
+    utils.createFile(fileTwoId, consignmentId)
+    val fileIds = Seq()
+
+    val files = fileRepository.getFileFields(fileIds).futureValue
+
+    files.size shouldBe 0
+    files shouldBe Seq()
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala:313]: No need to test if it is an empty `Seq()` as that's proven by checking the "size" is 0 above

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some very minor points. Thanks
